jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/AbstractUrlResourceFactory.java

```diff
@@ -0,0 +1,234 @@
+    {
+        return connectTimeout;
+    }
+
+    public void setConnectTimeout(int connectTimeout)
+    {
+        this.connectTimeout = connectTimeout;
+    }
+
+    public boolean isUseCaches()
+    {
+        return useCaches;
+    }
+
+    public void setUseCaches(boolean useCaches)
+    {
+        this.useCaches = useCaches;
+    }
+
+    @Override
+    public Resource newResource(final URI uri)
+    {
+        if (!uri.getScheme().equalsIgnoreCase(supportedProtocol))
+            throw new IllegalArgumentException("Scheme not support: " + uri.getScheme());
+
+        try
+        {
+            return new URLResource(uri, this.connectTimeout, this.useCaches);
+        }
+        catch (MalformedURLException e)
+        {
+            throw new RuntimeException("Bad URI: " + uri, e);
+        }
+    }
+
+    private static class URLResource extends Resource
+    {
+        private final URI uri;
+        private final URL url;
+        private final int connectTimeout;
+        private final boolean useCaches;
+
+        public URLResource(URI uri, int connectTimeout, boolean useCaches) throws MalformedURLException
+        {
+            this.uri = uri;
+            this.url = uri.toURL();
+            this.connectTimeout = connectTimeout;
+            this.useCaches = useCaches;
+        }
+
+        private URLConnection newConnection() throws IOException
+        {
+            URLConnection urlConnection = url.openConnection();
+            urlConnection.setUseCaches(this.useCaches);
+            urlConnection.setConnectTimeout(this.connectTimeout);
+            return urlConnection;
+        }
+
+        @Override
+        public Path getPath()
+        {
+            return null;
+        }
+
+        @Override
+        public boolean isContainedIn(Resource r)
+        {
+            // compare starting URIs?
+            return false;
+        }
+
+        @Override
+        public boolean isDirectory()
+        {
+            return uri.getPath().endsWith("/");
+        }
+
+        @Override
+        public boolean isReadable()
+        {
+            return exists();
+        }
+
+        @Override
+        public URI getURI()
+        {
+            return uri;
+        }
+
+        @Override
+        public String getName()
+        {
+            return uri.getPath();
+        }
+
+        @Override
+        public String getFileName()
+        {
+            return FileID.getFileName(uri);
+        }
+
+        @Override
+        public Resource resolve(String subUriPath)
+        {
+            URI newURI = uri.resolve(subUriPath);
+            try
+            {
+                return new URLResource(newURI, this.connectTimeout, this.useCaches);
+            }
+            catch (MalformedURLException e)
+            {
+                return null;
+            }
+        }
+
+        @Override
+        public boolean exists()
+        {
+            try
+            {
+                newConnection();
```

Comment:
Existence could probably be cached, i.e have a Boolean that is null if never tested and set by the first call to `newConnection()`... but then we should not worry too much about this class being efficient.

Reply:
If this class is used for `resourcebundle:` based URL protocols then we don't want existence cached.
If this class is used for `https:` then we also don't want the existence cached.

Besides, any kind of caching on our part would just conflict with the `URLConnection.setUseCaches(boolean)` behaviors.